src/precomputed_quantities.jl

```diff
@@ -147,6 +147,7 @@ function thermo_state(
     q_tot = nothing,
     q_pt = nothing,
 )
+    FT = eltype(thermo_params)
```

Comment:
I guess the allocation is from the FT here? Maybe I can just inline it? @charleskawczynski

Reply:
Yes I think so

Reply:
Hopefully that'll work, otherwise we may just need to revert this and remove the closures?

Reply:
it works:)